src/ensembl/src/content/app/entity-viewer/components/feature-length-axis/FeatureLengthAxis.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useEffect } from 'react';
+import { scaleLinear } from 'd3';
+
+import { getTicks } from './featureLengthAxisHelper';
+
+import styles from './FeatureLengthAxis.scss';
+
+import { getCommaSeparatedNumber } from 'src/shared/helpers/numberFormatter';
+
+type Ticks = {
+  ticks: number[];
+  labelledTicks: number[];
+};
+
+type Props = {
+  length: number; // number of biological building blocks (e.g. nucleotides) in the feature
+  width: number; // number of pixels allotted to the axis on the screen
+  onTicksCalculated?: (ticks: Ticks) => void; // way to pass the ticks to the parent if it is interested in them
+  standalone: boolean; // wrap the component in an svg element if true
+};
+
+const FeatureLengthAxis = (props: Props) => {
+  const domain = [1, props.length];
+  const range = [0, props.width];
+  const scale = scaleLinear()
+    .domain(domain)
+    .range(range);
+  const { ticks, labelledTicks } = getTicks(scale);
+
+  useEffect(() => {
+    props.onTicksCalculated &&
+      props.onTicksCalculated({ ticks, labelledTicks });
+  }, [props.length]);
+
+  const renderedAxis = (
+    <g>
+      <rect
+        className={styles.axis}
+        x={0}
+        y={0}
+        width={props.width}
+        height={1}
+      />
+      <text className={styles.label} x={0} y={20} textAnchor="end">
+        bp 1
+      </text>
+      {ticks.map((tick) => (
+        <g key={tick} transform={`translate(${scale(tick)})`}>
+          <rect className={styles.tick} key={tick} width={1} height={6} />
+          {labelledTicks.includes(tick) && (
+            <text className={styles.label} x={0} y={20} textAnchor="middle">
+              {getCommaSeparatedNumber(tick)}
+            </text>
+          )}
+        </g>
```

Comment:
Would it not be better to loop through and draw the labelled ticks separately instead of checking each ticks?

Reply:
Not sure what you mean. Labelled ticks are a subset of ticks, so when we are looping through the ticks we are also checking whether a given tick should also contain the label text next to it within the same `g` group. The x-coordinate of the tick (and the label) is defined by the `g` tag surrounding them both. Not sure what other approach you have in mind and what it will win (apart from some fewer loops caused by the `includes`, but they don't matter in this situation).

Reply:
Sorry, I thought that the labels can be drawn separately but didn't noice the wrapping `g` tag. It's good 👍